tephra-core/src/main/java/org/apache/tephra/TransactionManager.java

```diff
@@ -1378,17 +1403,32 @@ public TransactionType getTransactionType() {
     private final long expiration;
     private final InProgressType type;
     private final LongArrayList checkpointWritePointers;
+    private final String clientId;
```

Comment:
It is okay not to add it now since we are not persisting it. Add a comment in the code.